src/main/java/de/hpi/bpt/argos/core/Application.java

```diff
@@ -1,15 +1,49 @@
 package de.hpi.bpt.argos.core;
 
+import de.hpi.bpt.argos.eventHandling.EventSubscriber;
+import de.hpi.bpt.argos.persistence.database.DatabaseConnection;
+import de.hpi.bpt.argos.properties.PropertyEditor;
+import de.hpi.bpt.argos.properties.PropertyEditorImpl;
+
 /**
  * The main class that will be called on start of the application. Creates new argos.
  */
-public class Application {
+public final class Application {
+
+	/**
+	 * This method informs the user that this class should not be instantiated, as it is a utility class.
+	 */
+	private Application() {
+		throw new IllegalAccessError("Utility class");
+	}
+
 	/**
 	 * default java main method.
 	 * @param args - default parameter
 	 */
 	public static void main(String[] args) {
 		Argos argos = new ArgosImpl();
+
+		// edit properties if needed
+		PropertyEditor propertyEditor = new PropertyEditorImpl();
+		String eventPlatformHost = System.getProperty(EventSubscriber.getEventPlatformHostPropertyKey());
+		String databaseConnectionHost = System.getProperty(DatabaseConnection.getDatabaseConnectionHostPropertyKey());
+		String testMode = System.getProperty(Argos.getArgosBackendTestModePropertyKey());
+
+		// TODO: validate input in a more sophisticated way
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1135)